Revit_Core_Engine/Convert/Structure/FromRevit/Profile.cs

```diff
@@ -131,6 +131,84 @@ public static IProfile ProfileFromRevit(this FamilySymbol familySymbol, RevitSet
         }
 
+        /***************************************************/
+
+        [Description("Converts a Revit MullionType to BH.oM.Spatial.ShapeProfiles.IProfile.")]
+        [Input("mullionType", "Revit MullionType to be converted.")]
+        [Input("settings", "Revit adapter settings to be used while performing the convert.")]
+        [Input("refObjects", "Optional, a collection of objects already processed in the current adapter action, stored to avoid processing the same object more than once.")]
+        [Output("profile", "BH.oM.Spatial.ShapeProfiles.IProfile resulting from converting the input Revit MullionType.")]
+        public static IProfile ProfileFromRevit(this MullionType mullionType, RevitSettings settings = null, Dictionary<string, List<IBHoMObject>> refObjects = null)
+        {
+            settings = settings.DefaultIfNull();
+
+            IProfile profile = refObjects.GetValue<IProfile>(mullionType.Id);
+            if (profile != null)
+                return profile;
+
+            // The algorithm below is so convoluted because mullion type does not have a profile
+            // Instead, one needs to extract the geometry of one of the instances, and it needs to be the one that does not join with its start face
+            // First, take arbitrary mullion of a given type
+            Document doc = mullionType.Document;
+            Mullion representativeMullion = new FilteredElementCollector(doc)
+                .OfClass(typeof(FamilyInstance))
+                .Where(x => x is Mullion)
+                .Cast<Mullion>()
+                .FirstOrDefault(x => x.MullionType.Id.IntegerValue == mullionType.Id.IntegerValue && x.MullionLine() != null);
+
+            if (representativeMullion == null)
+            {
+                BH.Engine.Base.Compute.RecordError($"Profile could not be extracted from a mullion. ElementId: {mullionType.Id.IntegerValue}");
+                return null;
+            }
+
+            // Find the first mullion in the chain the arbitrary one belongs to
+            List<Mullion> allMullionsInWall = new FilteredElementCollector(doc)
+                .OfClass(typeof(FamilyInstance))
+                .Where(x => x is Mullion)
+                .Cast<Mullion>()
+                .Where(x => x.Host.Id.IntegerValue == representativeMullion.Host.Id.IntegerValue
+                         && x.MullionType.Id.IntegerValue == mullionType.Id.IntegerValue)
+                .ToList();
+
+            BH.oM.Geometry.Line line = representativeMullion.MullionLine();
+            BH.oM.Geometry.Vector direction = line.Direction();
+            foreach (Mullion candidateMullion in allMullionsInWall)
+            {
+                BH.oM.Geometry.Line candidateLine = candidateMullion.MullionLine();
+                if (candidateLine == null || !candidateLine.IsCollinear(line))
+                    continue;
+
+                if ((candidateLine.Start - line.Start).DotProduct(direction) > 0)
```

Comment:
What does it mean here when the dot product is > 0?

Reply:
I added a comment in the code 👍